plugin/pkg/shoot/seedmanager/admission.go

```diff
@@ -174,11 +174,28 @@ func determineSeed(shoot *garden.Shoot, seedLister gardenlisters.SeedLister, sho
 
 	// Determine all candidate seed cluster matching the shoot's cloud and region.
 	for _, seed := range seedList {
-		if seed.DeletionTimestamp == nil && seed.Spec.Cloud.Profile == shoot.Spec.Cloud.Profile && seed.Spec.Cloud.Region == shoot.Spec.Cloud.Region && seed.Spec.Visible != nil && *seed.Spec.Visible && verifySeedAvailability(seed) && hasDisjointedNetworks(seed, shoot) {
+		if seed.DeletionTimestamp == nil && seed.Spec.Cloud.Profile == shoot.Spec.Cloud.Profile && seed.Spec.Cloud.Region == shoot.Spec.Cloud.Region && seed.Spec.Visible != nil && *seed.Spec.Visible && verifySeedAvailability(seed) {
 			candidates = append(candidates, seed)
 		}
 	}
 
+	if candidates == nil {
+		return nil, errors.New("no adequate seed cluster found for this cloud profile and region")
+	}
+
+	old := candidates
```

Comment:
Should we use `old := append(candidates[:0:0], candidates...)` here (as stated [here](https://github.com/go101/go101/wiki/How-to-perfectly-clone-a-slice%3F))?
@adracus opinion?

Reply:
Good point!

Reply:
We actually don't copy the slice here. So no need to `append`.

Reply:
OK. Please squash the commits, then we can get it merged!

Reply:
Done!